mlflow/pytorch/torchserve/deploy/serve.py

```diff
@@ -0,0 +1,372 @@
+import docker
+import json
+import subprocess
+import time
+import logging
+import os
+import pandas as pd
+import torch
+from pathlib import Path, PurePath
+
+import requests
+from deploy.config import Config
+
+from mlflow.deployments import BaseDeploymentClient, get_deploy_client
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+
+_logger = logging.getLogger(__name__)
+
+_DEFAULT_TORCHSERVE_LOCAL_INFERENCE_PORT = "8080"
+_DEFAULT_TORCHSERVE_LOCAL_MANAGEMENT_PORT = "8081"
+
+
+class TorchServePlugin(BaseDeploymentClient):
+    def __init__(self, uri):
+
+        """
+        Initializes the deployment plugin and sets the environment variables
+        """
+        super(TorchServePlugin, self).__init__(target_uri=uri)
+        self.server_config = Config()
+        self.inference_api, self.management_api = self.__get_torch_serve_port()
+        self.default_limit = 100
+
+    def __get_torch_serve_port(self):
+        """
+        Reads through the config properties for torchserve inference and management api's
+        """
+        config_properties = self.server_config["config_properties"]
+        inference_port = "http://localhost:8080"
+        management_port = "http://localhost:8081"
+        address_strings = self.server_config["torchserve_address_names"]
+        if config_properties is not None and os.path.exists(config_properties):
+            with open(config_properties, "r") as f:
+                lines = f.readlines()
+                for line in lines:
+                    name = line.strip().split("=")
+                    if name[0] == address_strings[0] and name[1] is not None:
+                        inference_port = name[1]
+                    if name[0] == address_strings[1] and name[1] is not None:
+                        management_port = name[1]
+        return inference_port, management_port
+
+    def __validate_mandatory_arguments(self):
+        """
+        Validate the mandatory arguments is present if not raise exception
+        """
+
+        if not self.server_config["version"]:
```

Comment:
BTW, I think it'd be more consistent with other plugins to have the user pass these via the `config` argument to `mlflow deployments create` rather than environment variables - is it common for TorchServe users to pass these via environment variables? 

Also, are these variables actually mandatory? Looking at [TorchServe docs](https://pytorch.org/serve/model-archiver.html#arguments) it seems:
* We might be able to find the `model_file` within the logged model
* The `handler_file` is optional / torchserve has a default handler
* We may be able to infer a version for the model by listing existing models [via the REST API](https://pytorch.org/serve/management_api.html#describe-model) and creating a new model version equal to `previous_max_version + 1`

Note that we can always make required arguments optional later (e.g. in a follow-up PR or after the initial release of the plugin), but IMO it's probably worth it to see if we can supply as many reasonable defaults as possible to make it easy to use the plugin :)

Reply:
Thanks for your comments. We have committed new version of code which computes `VERSION` number and we have used config dictionary instead of environment variables. Tests and ReadMe files are updated as well.